hub/policy/src/main/java/uk/gov/ida/hub/policy/redis/SessionIdMixIn.java

```diff
@@ -0,0 +1,12 @@
+package uk.gov.ida.hub.policy.redis;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+/* The SessionId class is used for serialization both between microservices
+* and into Redis. To allow it to be serialized in the correct way for Redis,
+* this Mixin is included with the ObjectMapper Redis uses */
```

Comment:
I'm not sure I understand the point this comment is making.  Are you saying that:

 - SessionId is serialized both between microservices and for redis
 - SessionId is serialized in *different ways* between microservices and for redis
 - therefore, we can't use annotations to specify how to serialize SessionId in all cases
 - this MixIn specifies *just* the redis serialization behaviour

First of all, have I understood you correct?  Second, if so, can you answer some questions?

 - where is the JSON serialization for SessionId between microservices defined?
 - *why* is the serialization different between microservices and for redis?
 - can we make serialization the same so we can just use annotations like we do everywhere else?

Reply:
> * SessionId is serialized both between microservices and for redis

Yes, it is serialized into DTOs that saml-proxy sends and it is serialized in states that redis stores
> * SessionId is serialized in _different ways_ between microservices and for redis

Yes, Redis uses `@JsonCreator` and ignores getters, the microservices just use pure reflection
> * therefore, we can't use annotations to specify how to serialize SessionId in all cases

Adding the annotations breaks one of the serializations
> * this MixIn specifies _just_ the redis serialization behaviour

Yep

> * where is the JSON serialization for SessionId between microservices defined?

By providing empty constructors, Jackson uses reflection to work out how to populate the fields
> * _why_ is the serialization different between microservices and for redis?

Microservices uses and old and (comparatively) hard to reason about system
> * can we make serialization the same so we can just use annotations like we do everywhere else?

We _could_ but the problem is that the empty constructor/reflection method is used for *all* DTOs so it would require us to annotate every single one, including sub-objects. This is probably beneficial moving forwards but it seems out of scope for this work.

Reply:
We spoke face to face and we agree that we want SessionId to serialize as a raw string rather than as an object `{"sessionId":"foobarbaz"}`.  Hopefully this will mean we can get rid of the mixin?